### PR TITLE
Close featurelist (featureform) drawer on opening new project

### PR DESCRIPTION
**Pull request by signedav**
Be aware, if you are in the middle of editing a feature, this edits are lost. There is a error message in the log.
image
But IMO this is not a bad way of work. In case someone opens a project during editing a feature I guess he doesn't need to save that. And he get's informed as well...

### CODE REVIEW

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-544-1ca353f6fbd1f99216c8e93426884daf79d949a3-armv7.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-544-1ca353f6fbd1f99216c8e93426884daf79d949a3-x86.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-544-499a920e75849811763119d06e05ba1bd0cb854f-armv7.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-544-499a920e75849811763119d06e05ba1bd0cb854f-x86.apk)

**Comment by signedav** [comment]
Changed now, that it does cancel in case that it's in edit mode. There are no warnings anymore but the message "Changes discarded"...

Please test @mbernasocchi

File: src/qml/FeatureListForm.qml [hunk 1]
```diff
@@ -389,4 +388,19 @@ Rectangle {
       visible = false
     }
   }
+
+  //if project changed we should hide drawer
+  Connections {
+    target: iface
```
Comment (m-kuhn) [inline src/qml/FeatureListForm.qml]:
```suggestion
    target: currentLayer
```

File: src/qml/FeatureListForm.qml [hunk 2]
```diff
@@ -389,4 +388,19 @@ Rectangle {
...
+
+    onLoadProjectEnded: {
```
Comment (m-kuhn) [inline src/qml/FeatureListForm.qml]:
```suggestion
    onDestroyed: {
```

**Comment by m-kuhn** [comment]
@signedav I think this proposed approach might be more generic, but I'm not sure if it works.

**Comment by signedav** [comment]
@m-kuhn I had no success with your approach.

What it's doing now with my solution: It pedals back out of the form without doing anything. To not doing anything it needs to "simulate" a cancel (because on closing without cancel it tries to save) but does not reset the model.

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-544-cb6c428d31da5b8d207cb526231caf13f3e5e138-armv7.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-544-cb6c428d31da5b8d207cb526231caf13f3e5e138-x86.apk)

**Comment by m-kuhn** [comment]
> Like this:
> _MG_3639

That sounds like the way it needs to be. 👍

**Pull request merged**